src/dask_awkward/lib/structure.py

```diff
@@ -506,40 +514,55 @@ def zip(
     if not highlevel:
         raise ValueError("Only highlevel=True is supported")
 
-    if not isinstance(arrays, dict):
-        raise DaskAwkwardNotImplemented(
-            "dask_awkward.zip only supports dictionary inputs."
+    if isinstance(arrays, dict):
+        keys, colls, metadict = [], [], {}
+        for k, coll in arrays.items():
+            keys.append(k)
+            colls.append(coll)
+            metadict[k] = coll._meta
+
+        meta = ak.zip(
+            metadict,
+            depth_limit=depth_limit,
+            parameters=parameters,
+            with_name=with_name,
+            highlevel=highlevel,
+            behavior=behavior,
+            right_broadcast=right_broadcast,
+            optiontype_outside_record=optiontype_outside_record,
         )
 
-    keys, colls, metadict = [], [], {}
-    for k, coll in arrays.items():
-        keys.append(k)
-        colls.append(coll)
-        metadict[k] = coll._meta
-
-    meta = ak.zip(
-        metadict,
-        depth_limit=depth_limit,
-        parameters=parameters,
-        with_name=with_name,
-        highlevel=highlevel,
-        behavior=behavior,
-        right_broadcast=right_broadcast,
-        optiontype_outside_record=optiontype_outside_record,
-    )
+        return map_partitions(
+            _ZipDictInputFn(
+                keys,
+                depth_limit=depth_limit,
+                parameters=parameters,
+                with_name=with_name,
+                highlevel=highlevel,
+                behavior=behavior,
+                right_broadcast=right_broadcast,
+                optiontype_outside_record=optiontype_outside_record,
+            ),
+            *colls,
+            label="zip",
+            meta=meta,
+        )
 
-    return map_partitions(
-        _ZipFn(
-            keys,
+    elif isinstance(arrays, list):
```

Comment:
Why is this restricted to `list`s here? This precludes other iterables that should work (like `tuple`s)